compiler/qlc.ml

```diff
@@ -0,0 +1,12 @@
+type action = Raw | Ast | Semantic | Semantic_to_jast | Code_generator
```

Comment:
can you elaborate on why we moved this stuff into a new file? i'm assuming this is related to how we are now compiling things differently?